linux_os/guide/services/sssd/sssd-ldap/sssd_ldap_start_tls/ansible/shared.yml

```diff
@@ -18,14 +18,27 @@
   changed_when: False
 
 - name: "Add default domain group and use STARTTLS (if no domain there)"
-  lineinfile:
+  ini_file:
     path: /etc/sssd/sssd.conf
-    line: "[domain/default]\nldap_id_use_start_tls = True\n"
+    section: domain/default
+    section: "{{ item.section }}"
+    option: "{{ item.option }}"
+    value: "{{ item.value }}"
+    create: yes
+    mode: 0600
+  with_items:
+    - { section: sssd, option: domains, value: default}
+    - { section: domain/default, option: id_provider, value: files }
+    - { section: domain/default, option: ldap_id_use_start_tls, value: true}
   when: test_grep_domain.stdout == ""
 
 - name: "Configure LDAP to use STARTTLS"
-  lineinfile:
+  ini_file:
     path: /etc/sssd/sssd.conf
-    regexp: '^\s*ldap_id_use_start_tls'
-    insertafter: '\s*\[domain\/[^]]*]'
-    line: 'ldap_id_use_start_tls = True'
+    section: "{{ test_grep_domain.stdout | regex_replace('[(.*)]','\\1') }}"
```

Comment:
I'm receiving the following error:
```
< TASK [Configure LDAP to use STARTTLS] >
 ---------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

An exception occurred during task execution. To see the full traceback, use -vvv. The error was: re.error: invalid group reference 1 at position 1                                                                
fatal: [all]: FAILED! => {"msg": "Unexpected failure during module execution.", "stdout": ""}
        to retry, use: --limit @/home/jcerny/scap-security-guide/sssd.retry
```

Reply:
The square brackets needed to be esaped. Fixed